core-bundle/src/Resources/contao/library/Contao/FeedItem.php

```diff
@@ -113,12 +118,23 @@ public function addEnclosure($strFile, $strUrl=null, $strMedia='enclosure')
 			$strUrl = Environment::get('base');
 		}
 
+		$fileUrl = $strUrl . System::urlEncode($strFile);
 		$objFile = new File($strFile);
 
+		$size = StringUtil::deserialize($imageSize, true);
+
+		if ($size && $objFile->isImage)
+		{
+			$image = System::getContainer()->get('contao.image.image_factory')->create($rootDir . '/' . $strFile, $size);
+			$fileUrl = $strUrl . System::urlEncode($image->getUrl($rootDir));
+			$relativeFilePath = Path::makeRelative($image->getPath(), $rootDir);
+			$objFile = new File($relativeFilePath);
```

Comment:
Are you sure to use `File` class for a non-resized image?

Reply:
I don’t understand the question. It overwrites the `$objFile` variable from above which seems to be correct.

Reply:
As far as I see the file is not required to be present, when instantiating `File`? However, I added a check if the file exists, before I get `$objFile->size` some lines below.